pkg/cloud/ibmcloud/deployer.go

```diff
@@ -75,6 +75,11 @@ func (d *DeploymentClient) GetKubeConfig(cluster *clusterv1.Cluster, master *clu
 	sshUserName := providerSpec.SshUserName
 
 	privateKey := "id_ibmcloud"
+	envPrivateKey, ok := os.LookupEnv("IBMCLOUD_HOST_SSH_PRIVATE_FILE")
+	if ok && (0 != strings.Compare(envPrivateKey, "")) {
+		klog.V(3).Infof("Found environment variable to SSH private file %s", envPrivateKey)
+		privateKey = envPrivateKey
+	}
 
 	result := strings.TrimSpace(util.ExecCommand(
 		"ssh", "-i", homeDir+"/.ssh/"+privateKey,
```

Comment:
I might be overkill here
just curious as we are handling customrize anyway now, why we still need homeDir+/.ssh/+ the given file
instead use given file location anyway?

Reply:
Agree, env variable should be support absolute path. I will try 2 cases.
absolute file and relative file. Thanks

Reply:
Update code to support absolute path and relative path for the ssh key files.

Reply:
Thanks for @jichenjc's comments.